### PR TITLE
Report TestNG test methods below test classes

### LINKED ISSUE #6679: Recent TestNG TestDescriptor event structure looks suspicious
See [this email thread](https://groups.google.com/a/gradle.com/forum/#!topic/gradle-enterprise-team/pDVn5d-X14A), and this [dotcom issue](https://github.com/gradle/dotcom/issues/1402)

Basically a new TestDescriptor event seems to be generated by Gradle, corresponding to a 'class level suite event', but not properly assigned in the hierarchy of events.

We capture this event in build scan too. It's affected our handling of TestNG tests as it is tough.
Yet, I'm raising this issue because it doesn't seem right.

### Expected Behavior
```
gradle test executor test descriptor
     test Ng suite test descriptor
        class level suite test descriptor (*new event*)
            test method test descriptor
```
The new added event is added as a parent of test method events.

### Current Behavior
```
gradle test executor test descriptor
     test Ng suite test descriptor
        class level suite test descriptor (*new event*)
        test method test descriptor
```
The new added event is added as a sibling of test method events. Shouldn't it be a parent ?

### Context
I was checking if new TestNG versions were handled properly in Gradle Enterprise.
They are, as we luckily ignore this new test descriptor because it doesn't have 'test method test descriptor' as children.
If and when this is fixed in the Build Tool, we'll need to update our handling in GE, too.

### Steps to Reproduce (for bugs)
```
apply plugin: 'java'
dependencies {
    testCompile 'org.testng:testng:6.9.11'
    }
test.useTestNG()
```

Then, any TestNG example should exhibit the problem. Or you can use a test class in `src/test/java/org/company/SystemOutTest.java`
```
package org.company;
        import org.testng.Assert;
        import org.testng.annotations.Test;
        @Test
        public class SystemOutTest {
          @Test
          public void testOut() {
            System.out.println("System.out rules!");
            Assert.assertTrue(true);
          }
        }
```

### PR DESCRIPTION
**Pull request by marcphilipp**
Prior to this commit, test methods were reported as siblings of their
test classes for recent versions of TestNG:

```
gradle test executor test descriptor
└── test Ng suite test descriptor
    ├── class level suite test descriptor
    └── test method test descriptor
```

Now, test methods are reported as children of their test classes:

```
gradle test executor test descriptor
└── test Ng suite test descriptor
    └── class level suite test descriptor
        └── test method test descriptor
```

Resolves #6679.

### CODE REVIEW

**Comment by marcphilipp** [comment]
Green build: https://builds.gradle.org/viewLog.html?buildId=15934850&tab=buildResultsDiv&buildTypeId=Gradle_Check_Stage_ReadyforMerge_Trigger

**Comment by etiennestuder** [comment]
@facewindu Will this affect us?

**Comment by marcphilipp** [comment]
Yes, it will (@facewindu originally reported #6679).

File: gradle/dependencies.gradle [hunk 1]
```diff
@@ -83,7 +83,7 @@ libraries.pmaven_groovy =       [coordinates: 'org.sonatype.pmaven:pmaven-groovy
 libraries.rhino =               [coordinates: 'org.mozilla:rhino', version: '1.7.10']
 libraries.simple =              [coordinates: 'org.simpleframework:simple', version: '4.1.21']
 libraries.snakeyaml =           [coordinates: 'org.yaml:snakeyaml', version: '1.17']
-libraries.testng =              [coordinates: 'org.testng:testng', version: '6.3.1']
+libraries.testng =              [coordinates: 'org.testng:testng', version: '6.3.1', because: 'later versions break test cross-version test filtering']
```
[the remark this answers is not in the document]
Comment (marcphilipp) [reply gradle/dependencies.gradle]: It's most likely a binary incompatibility between different versions.

File: subprojects/testing-jvm/src/main/java/org/gradle/api/internal/tasks/testing/testng/TestNGTestResultProcessorAdapter.java [hunk 2]
```diff
@@ -97,6 +97,9 @@ public void onBeforeClass(ITestClass testClass) {
             testInternal = new DefaultTestClassDescriptor(idGenerator.generateId(), testClass.getName());
             testClassId.put(testClass, testInternal.getId());
             parentId = xmlTestIds.get(testClass.getXmlTest());
+            for (ITestNGMethod method : testClass.getTestMethods()) {
+                testMethodParentId.put(method, testInternal.getId());
```
[the remark this answers is not in the document]
Comment (marcphilipp) [reply subprojects/testing-jvm/src/main/java/org/gradle/api/internal/tasks/testing/testng/TestNGTestResultProcessorAdapter.java]: Older TestNG versions don't support `onBeforeClass()` and thus never call it. Therefore, the code in `onStart()` sets the preliminary parents, assuming `onBeforeClass()` might never get called.

**Pull request merged**